app/domain/authentication/authn_jwt/restriction_validation/create_constraints_from_policy.rb

```diff
@@ -0,0 +1,104 @@
+require 'command_class'
+
+module Authentication
+  module AuthnJwt
+    module RestrictionValidation
+      CreateConstrainsFromPolicy = CommandClass.new(
+        dependencies: {
+          non_permitted_constraint_class: Authentication::Constraints::NonPermittedConstraint,
+          required_constraint_class: Authentication::Constraints::RequiredConstraint,
+          multiple_constraint_class: Authentication::Constraints::MultipleConstraint,
+          not_empty_constraint: Authentication::Constraints::NotEmptyConstraint.new,
+          fetch_mandatory_claims: Authentication::AuthnJwt::RestrictionValidation::FetchMandatoryClaims.new,
+          fetch_mapping_claims: Authentication::AuthnJwt::RestrictionValidation::FetchMappingClaims.new,
+          logger: Rails.logger
+        },
+        inputs: %i[authentication_parameters non_permitted_constraints]
+      ) do
+        def call
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreateContraintsFromPolicy.new)
+          init_constraints_list
+          add_non_empty_constraint
+          fetch_mandatory_claims
+          fetch_claims_mapping
+          map_mandatory_claims
+          add_required_constraint
+          add_non_permitted_constraint
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreatedConstraintsFromPolicy.new)
```

Comment:
i think the debug log should be after the action `create_multiple_constraint`
and you can put in the end of the function what you are planning to return : `multiple_constraint`